frontend_app_gallery/src/component/MainPreview.jsx

```diff
@@ -1,23 +1,31 @@
 import React from 'react'
 import PropTypes from 'prop-types'
 import classnames from 'classnames'
+import { translate } from 'react-i18next'
+import { IMG_LOAD_STATE } from '../helper.js'
 
-class MainPreview extends React.Component {
+export class MainPreview extends React.Component {
   constructor (props) {
     super(props)
 
     this.state = {
       height: 0,
       width: 0,
-      imageLoaded: false
+      imageLoaded: IMG_LOAD_STATE.LOADING
     }
   }
 
-  onLoad ({ target: img }) {
+  onLoad = ({ target: img }) => {
```

Comment:
this naming is too generic, it existed before but now is a good time to rename it
proposition: onImageLoad